### ISSUE TITLE
Unexpected migration issue while spawning archival node on testnet

### ISSUE THREAD

**Issue by miraclx** [near/nearcore#4624]
```console
$ neard run
Aug 02 13:14:54.505  INFO neard: Version: trunk, Build: 61584bba, Latest Protocol: 46                                                                                                                                                       
Aug 02 13:17:49.769  INFO near: Opening store database at "/home/ubuntu/.near/data"                                                                                                                                                         
Aug 02 13:17:49.890  INFO near: Migrate DB from version 24 to 25                                                                                                                                                                            
thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: Custom { kind: InvalidInput, error: "Unexpected variant index: 15" }', nearcore/src/migrations.rs:448:79                                                            
```

**To Reproduce**
Spin up an *archival* node on master (b95e753 at the time of writing) on testnet.

**Expected behavior**
DB Migration from 24 to 25 doesn't panic, node is spawned successfully

**Context**
Stumbled on this while investigating #4600, verified it's the same behavior on master.

cc. @bowenwang1996 @ailisp

*This migration was introduced in #4466*

**Comment by frol** [comment]
@khorolets FYI, we should not even consider migrating to some latest nearcore commit in indexer since that will break on migration on our archival nodes.

@janewang @chefsale we should consider canary archival nodes (at least checking that those are capable to boot up) since this time it caught me and @miraclx off guard.

**Comment by bowenwang1996** [comment]
@ailisp please take a look

**Comment by frol** [comment]
@bowenwang1996 I believe we should move 1.21-rc release from the next Monday since @ailisp is on vacation this week, so he won't be able to investigate this issue.

**Comment by bowenwang1996** [comment]
@frol I see. Thanks for the reminder!

**Comment by ailisp** [comment]
#4466 fixed the corrupt execution outcome in mainnet archival node and boot mainnet archival node. But this shows testnet node archival node has outcomes corrupt in a different way, I'm investigating how they're corrupted and how to correct them

**Comment by ailisp** [comment]
I made a fix in #4662 and explained how it is corrupted there
